File: DuDoong-Api/src/main/java/band/gosrock/api/issuedTicket/dto/response/RetrieveIssuedTicketDetailResponse.java
package band.gosrock.api.issuedTicket.dto.response;


import band.gosrock.domain.common.vo.EventInfoVo;
import band.gosrock.domain.common.vo.IssuedTicketInfoVo;
import band.gosrock.domain.domains.event.domain.Event;
import band.gosrock.domain.domains.issuedTicket.domain.IssuedTicket;
import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
@Builder
public class RetrieveIssuedTicketDetailResponse {

    /*
    발급 티켓 정보 VO
     */
    private final IssuedTicketInfoVo issuedTicketInfo;

    /*
    이벤트 정보 VO
     */
    private final EventInfoVo eventInfo;

    public static RetrieveIssuedTicketDetailResponse of(IssuedTicket issuedTicket, Event event) {
        return RetrieveIssuedTicketDetailResponse.builder()
                .issuedTicketInfo(issuedTicket.toIssuedTicketInfoVo())
            .eventInfo(event.toEventInfoVo())
                .build();
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/issuedTicket/dto/response/RetrieveIssuedTicketDetailResponse.kt
package band.gosrock.api.issuedTicket.dto.response

import band.gosrock.domain.common.vo.EventInfoVo
import band.gosrock.domain.common.vo.IssuedTicketInfoVo
import band.gosrock.domain.domains.event.domain.Event
import band.gosrock.domain.domains.issuedTicket.domain.IssuedTicket

data class RetrieveIssuedTicketDetailResponse(
    val issuedTicketInfo: IssuedTicketInfoVo,
    val eventInfo: EventInfoVo,
) {
    companion object {
        @JvmStatic
        fun of(issuedTicket: IssuedTicket, event: Event): RetrieveIssuedTicketDetailResponse {
            return RetrieveIssuedTicketDetailResponse(
                issuedTicketInfo = issuedTicket.toIssuedTicketInfoVo(),
                eventInfo = event.toEventInfoVo(),
            )
        }
    }
}
